compaction/compaction.cc

```diff
@@ -1472,11 +1481,18 @@ class scrub_compaction final : public regular_compaction {
         return _options.operation_mode == compaction_type_options::scrub::mode::segregate;
     }
 
+    compaction_result finish(std::chrono::time_point<db_clock> started_at, std::chrono::time_point<db_clock> ended_at) override {
+        auto ret = compaction::finish(started_at, ended_at);
+        ret.validation_errors = _cts->validation_errors;
```

Comment:
If we use `compaction_result` to communicate back validation error count, why do we need a `_cts` passed from outside? Can't we just have a local counter instead?

Reply:
What do you mean by local?
We need to keep that as a member of `scrub_compaction` because during a single compaction many instances of reader are created with `make_sstable_reader()` and they need to share the same counter. I think it's clearer than passing a local variable through all the functions.

Reply:
I mean add a `uint64_t _validation_errors` member to `scrub_compaction` and we just pass reference to this member to scrub readers. Then in `finish()` we just do:
```
ret.validation_errors = _validation_errors;
```